Missed k8s.io/kube-openapi/cmd/openapi-gen dependency on code-generator go.mod

### What happened?

We're trying to upgrade `k8s.io/code-generator` to `v1.30.1`  (https://github.com/kubernetes-sigs/kueue/pull/2402) and found the issue on executing `kube::codegen::gen_openapi` (https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/2402/pull-kueue-verify-main/1800788997593632768). 

```
../kube-openapi@v0.0.0-20240228011516-70dd3763d340/pkg/common/common.go:23:2: missing go.sum entry for module providing package github.com/emicklei/go-restful/v3 (imported by k8s.io/kube-openapi/pkg/common); to add:
	go get k8s.io/kube-openapi/pkg/common@v0.0.0-20240228011516-70dd3763d340
```

The problem is that on go.mod (https://github.com/kubernetes/code-generator/blob/master/go.mod) missed `github.com/emicklei/go-restful/v3` dependency that not allow to install `k8s.io/kube-openapi` (https://github.com/kubernetes/code-generator/blob/master/kube_codegen.sh#L330C45-L330C64)

### What did you expect to happen?

No error

### How can we reproduce it (as minimally and precisely as possible)?

To reproduce the same what happening on https://github.com/kubernetes/code-generator/blob/master/kube_codegen.sh#L325-L330, just execute this command on code-generator folder:
```bash
GO111MODULE=on go install k8s.io/kube-openapi/cmd/openapi-gen
```

### Anything else we need to know?

I think it's happens after this commit https://github.com/kubernetes/kubernetes/commit/1a4450f6ffdad44bfc12fae5911c93e3bd238f0f.

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>


### Cloud provider

<details>

</details>


### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>


### Install tools

<details>

</details>


### Container runtime (CRI) and version (if applicable)

<details>

</details>


### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

I was able to reproduce, but just because I had an older version of kube-openapi in my GOPATH. Using the `@latest` prefix in the command fixed it.

```
$ GO111MODULE=on go install k8s.io/kube-openapi/cmd/openapi-gen
../../gopath/pkg/mod/k8s.io/kube-openapi@v0.0.0-20240228011516-70dd3763d340/pkg/common/common.go:23:2: missing go.sum entry for module providing package github.com/emicklei/go-restful/v3 (imported by k8s.io/kube-openapi/pkg/common); to add:
        go get k8s.io/kube-openapi/pkg/common@v0.0.0-20240228011516-70dd3763d340
$ GO111MODULE=on go install k8s.io/kube-openapi/cmd/openapi-gen@latest
go: downloading k8s.io/kube-openapi v0.0.0-20240521193020-835d969ad83a
$ GO111MODULE=on go install k8s.io/kube-openapi/cmd/openapi-gen@latest
```

I think line https://github.com/kubernetes/code-generator/blob/master/kube_codegen.sh#L330C45-L330C64 needs to be

```
GO111MODULE=on go install $(printf "k8s.io/kube-openapi/cmd/%s@latest" "${BINS[@]}")
```

--- a/staging/src/k8s.io/code-generator/go.mod
+++ b/staging/src/k8s.io/code-generator/go.mod
@@ -20,6 +20,7 @@ require (
 )
 
 require (
+	github.com/emicklei/go-restful/v3 v3.11.0 // indirect
 	github.com/go-logr/logr v1.4.1 // indirect
 	github.com/go-openapi/jsonpointer v0.19.6 // indirect
 	github.com/go-openapi/jsonreference v0.20.2 // indirect

--- a/staging/src/k8s.io/code-generator/tools.go
+++ b/staging/src/k8s.io/code-generator/tools.go
@@ -31,4 +31,5 @@ import (
 	_ "k8s.io/code-generator/cmd/informer-gen"
 	_ "k8s.io/code-generator/cmd/lister-gen"
 	_ "k8s.io/code-generator/cmd/register-gen"
+	_ "k8s.io/kube-openapi/cmd/openapi-gen"
 )

